### DISCUSSION TITLE
Question about Handling Information Between Nested Grids in MOOSE

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#26623] [Q&A General] [answered]
Hi MOOSE community,

I've been working on a problem involving two nested grids recently. I've learned that CoarseMeshExtraElementIDGenerator can help establish a mapping between coarse and fine grid elements. However, besides the mapping between elements, I'm interested in preserving some additional information about the coarse grid.

I'm curious about whether the coarse grid can still be accessed after leaving the Mesh block. Specifically, I'd like to know if its memory is released, and if MOOSE only retains information about the fine grid.

Additionally, I'm wondering if there's a way to pass data structures, like std::vector, between different kernels in MOOSE. I'm considering saving the coarse grid information in a vector or another suitable form and then inputting it into the required object. If using std::vector directly, I'm unsure about how to declare it in the input file (input.i). I've thought about creating a custom user object to store this information, but I'm not sure how to handle its execute and other functions since I don't need it to perform any calculations.

Any insights or guidance on these questions would be greatly appreciated!

Thank you,
hityyds

**Comment by GiudGiud** [comment, marked as answer]
Hello

There is a save_in parameter in the mesh generator blocks to let you save another mesh.
Then the memory wont be released and you can access it during the simulation, 

> Additionally, I'm wondering if there's a way to pass data structures, like std::vector, between different kernels in MOOSE

You can either
- save data to tagged vectors and matrices but it s not an arbitrary data structure. 
- use a material property to compute the quantity you need and share it that way

But for the coarse grid, looking at save_in is probably the way

Guillaume

**Reply by hityyds** [reply]
Thank you very much for your detailed explanation. Additionally, do you know how to make an object able to load a mesh? I've tried to learn the meshgenerator code, but it includes so many files that I can't figure out which ones I need to include in order to have the functional interfaces.

**Reply by GiudGiud** [reply]
look at the test object
meshgenerators/TestMeshGenerator.C

and getMeshByName

**Reply by hityyds** [reply]
Thanks again for your help.
